tests/suites/test_suite_ssl.function

```diff
@@ -1789,18 +1839,55 @@ void perform_handshake( handshake_test_options* options )
     }
 #endif
 
-    TEST_ASSERT( mbedtls_move_handshake_to_state( &(client.ssl),
-                                                  &(server.ssl),
-                                                  MBEDTLS_SSL_HANDSHAKE_OVER )
-                 == 0 );
+    if( options->use_min_max_version &&
```

Comment:
Could we please make this more general? I mean we could just have an expected return value in the options and we would check here if we are getting the return value we expected. If it we expected failure, then we can then jump to cleanup.

Reply:
The difficulty here is that we already have an expected negotiated version - I didn't particularly want to add another field as this function already has more than most, if not all. I have attempted to at least make this code more neat (less duplication).

Reply:
I like the solution you found for this problem.